src/database/DataTypes/StocktakeBatch.js

```diff
@@ -211,7 +211,25 @@ export class StocktakeBatch extends Realm.Object {
    */
   set countedTotalQuantity(quantity) {
     // Handle packsize of 0.
-    this.countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+    const countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+
+    this.countedNumberOfPacks = countedNumberOfPacks;
+    this.doses = countedNumberOfPacks * this.dosesPerVial;
+  }
+
+  get isVaccine() {
+    return this.itemBatch?.item?.isVaccine ?? false;
+  }
+
+  get dosesPerVial() {
+    return this.isVaccine ? this.itemBatch?.item?.doses ?? 0 : 0;
+  }
+
+  setDoses(database, newValue) {
```

Comment:
`setCountedDoses`?

Reply:
i'll keep it as `setDoses` since the field is `doses` but will try and argue more to change `item` to `dosesPerVial` and will try go for `countedDoses` also